src/app/containers/ArticleTimestamp/helpers.test.js

```diff
@@ -1,20 +1,137 @@
-      expect(isValidDateTime()).toEqual(false);
+      expect(isValidDateTime('foo')).toBe(false);
+      expect(isValidDateTime(null)).toBe(false);
+      expect(isValidDateTime(undefined)).toBe(false);
+      expect(isValidDateTime()).toBe(false);
+      expect(isValidDateTime(NaN)).toBe(false);
+    });
+  });
+
+  describe('isFirstRelative', () => {
+    describe(`when lastPublished === firstPublished`, () => {
+      const timeLimits = generateTimeLimits(15);
+      timeLimits.forEach(({ expectedValue, time, title }) => {
+        it(`should return ${expectedValue} for ${title}`, () => {
+          const firstPublished = time;
+          const lastPublished = time;
+          expect(isFirstRelative(lastPublished, firstPublished)).toBe(
+            expectedValue,
+          );
+        });
+      });
+    });
+
+    it(`should return false when lastPublished !== firstPublished`, () => {
+      const firstPublished = timestampGenerator({ hours: 2 });
+      const lastPublished = timestampGenerator({ days: 3 });
+      expect(isFirstRelative(lastPublished, firstPublished)).toBe(false);
+    });
+  });
+
+  describe('isLastRelative', () => {
+    const timeLimits = generateTimeLimits(15);
+    timeLimits.forEach(({ expectedValue, time, title }) => {
+      it(`should return ${expectedValue} when lastPublished is ${title}`, () => {
+        const lastPublished = time;
+        expect(isLastRelative(lastPublished)).toBe(expectedValue);
+      });
+    });
+  });
+
+  describe('isSameDay', () => {
+    it('should return true if both timestamps are on the same day', () => {
+      const firstPublished = 1569962265000;
+      const lastPublished = 1569962145008;
+      expect(isSameDay(firstPublished, lastPublished)).toBe(true);
+    });
+
+    it('should return false if both timestamps are not on the same day', () => {
+      const fourHoursAgo = timestampGenerator({ hours: 4 });
+      const thirtySixHoursAgo = timestampGenerator({ hours: 36 });
+      expect(isSameDay(fourHoursAgo, thirtySixHoursAgo)).toBe(false);
+    });
+  });
+
+  describe('isToday', () => {
+    it('should return true if timestamp is today (3 minutes ago)', () => {
+      const timestamp = timestampGenerator({ minutes: 3 });
+      expect(isToday(timestamp)).toBe(true);
+    });
+
+    it('should return false if timestamp is not today (32 hours ago)', () => {
+      const timestamp = timestampGenerator({ hours: 32 });
+      expect(isToday(timestamp)).toBe(false);
+    });
+  });
+
+  describe('formatType', () => {
+    const dateFormats = {
+      date: 'D MMMM YYYY',
+      dateTimeTimezone: 'D MMMM YYYY, HH:mm z',
+    };
+    it(`should return default date format when firstPublished is not today`, () => {
+      const firstPublished = timestampGenerator({ days: 5 });
+      expect(formatType({ firstPublished })).toBe(dateFormats.date);
+    });
+
+    it(`should return dateTimeTimezone format when firstPublished is today`, () => {
```

Comment:
"dateTimeTimezone" has no meaning outside these tests, could we say "date and time with timezone format"? (Here and below)